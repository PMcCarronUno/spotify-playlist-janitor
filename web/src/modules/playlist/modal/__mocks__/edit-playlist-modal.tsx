import React from 'react';
export const EditPlaylistModalView = () => <div>Edit Playlist Modal View</div>