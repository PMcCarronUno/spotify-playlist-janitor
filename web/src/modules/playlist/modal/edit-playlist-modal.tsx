import styled from "styled-components";
import { RED } from "shared/constants";
import { SecondaryText, SubTitle, Text } from "shared/components/typography";
import { PrimaryButton, SecondaryButton } from "shared/components/button";
import { TextInput } from "shared/components/text-input";
import { Tooltip } from "shared/components/tooltip";
import { VscQuestion } from "react-icons/vsc";
import { Toggle } from "shared/components/toggle";
import { DetailedPlaylist } from "../playlist-types";

type EditPlaylistModalViewProps = {
  onSubmit(event: React.FormEvent<HTMLFormElement>): Promise<void>;
  playlist: DetailedPlaylist | undefined;
  modalSaving: boolean;
  modalError?: string;
  onModalClose(): void;
};

export const EditPlaylistModalView = ({
  onSubmit,
  playlist,
  modalSaving,
  modalError,
  onModalClose,
}: EditPlaylistModalViewProps) => {
  return (
    <form {...{ onSubmit }} autoComplete="off">
      <ModalContainer>
        <SubTitle style={{ marginBottom: 8 }}>
          Edit playlist details
        </SubTitle>
        <div className="rows">
          <div className="row">
            <Text>Skip threshold (seconds):</Text>
            <TextInput
              {...{
                className: "number-input",
                id: "skip-threshold-input",
                "data-testid": "skip-threshold-input",
                type: "number",
                defaultValue: playlist?.skipThreshold,
                min: 1,
                max: 999,
                varaint: "boxed",
                disabled: modalSaving,
              }}
            />
            <Tooltip
              content={
                "The track progress before which a change will be counted as a skip."
              }
              dataTooltipId="skip-threshold-tooltip"
            >
              <VscQuestion />
            </Tooltip>
          </div>
          <div className="row">
            <div className="toggle-input">
              <Toggle
                {...{
                  label: "Ignore initial skips",
                  id: "ignore-intial-skips-toggle",
                  "data-testid": "ignore-intial-skips-toggle",
                  disabled: modalSaving,
                  defaultChecked: playlist?.ignoreInitialSkips
                }}
              />
            </div>
            <Tooltip
              content={
                "Ignore any skips that occur when listening first begins until a song has exceeded the Skip Threshold, any skip after that will be counted."
              }
              dataTooltipId="initial-skips-tooltip"
            >
              <VscQuestion />
            </Tooltip>
          </div>
          <div className="row">
            <Text>Auto-delete tracks after:</Text>
              <TextInput
                {...{
                  className: "number-input",
                  id: "auto-delete-input",
                  "data-testid": "auto-delete-input",
                  type: "number",
                  min: 0,
                  max: 999,
                  varaint: "boxed",
                  disabled: modalSaving,
                  defaultValue: playlist?.autoCleanupLimit
                }}
              />
            <Text className="auto-delete-post-text">skips</Text>
            <Tooltip
              content={
                "Can be left blank to prevent automatic deletion of any tracks."
              }
              dataTooltipId="aut-delete-tooltip"
            >
              <VscQuestion />
            </Tooltip>
          </div>
          {modalError && (
            <div
              {...{
                className: "row error",
                id: "modal-error",
                "data-testid": "modal-error",
              }}
            >
              <SecondaryText>{modalError}</SecondaryText>
            </div>
          )}
        </div>
        <div
          style={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "flex-end",
          }}
        >
          <SecondaryButton
            {...{
              className: "secondary",
              style: { margin: "0px 4px" },
              type: "button",
              id: "cancel-modal-button",
              "data-testid": "cancel-modal-button",
              onClick: onModalClose,
              disabled: modalSaving,
            }}
          >
            Cancel
          </SecondaryButton>
          <PrimaryButton
            {...{
              className: "primary",
              style: { margin: "0px 4px" },
              type: "submit",
              id: "confirm-modal-button",
              "data-testid": "confirm-modal-button",
              disabled: modalSaving,
            }}
          >
            Confirm
          </PrimaryButton>
        </div>
      </ModalContainer>
    </form>
  );
};

const ModalContainer = styled.div`
  width: 360px;

  .rows {
    margin-bottom: 8px;
  }

  .row {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-bottom: 8px;
  }

  .number-input {
    width: 55px !important;
  }

  .toggle-input {
    margin-right: 10px;
  }

  .auto-delete-post-text {
    margin-right: 10px;
  }

  .error {
    color: ${RED};
  }
`;

EditPlaylistModalView.displayName = "EditPlaylistModalView";
