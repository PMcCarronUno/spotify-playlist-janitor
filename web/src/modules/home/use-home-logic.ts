import { useCallback, useEffect, useState } from "react";

import { useDataApi } from "api/data-api";
import { useSpotifyApi } from "api/spotify-api";

import { Playlist } from "./home-types";
import { useModal } from "shared/components/modal";

export const useHomeLogic = () => {
  const { getDatabasePlaylists, addDatabasePlaylist } = useDataApi();
  const { getSpotifyPlaylists } = useSpotifyApi();

  const [loading, setLoading] = useState(false);
  const [modalSaving, setModalSaving] = useState(false);
  const [showSpotifyAuthModal, setShowSpotifyAuthModal] = useState(false);
  const [monitoredPlaylists, setMonitoredPlaylists] = useState<
    Playlist[] | undefined
  >(undefined);
  const [unmonitoredPlaylists, setUnmonitoredPlaylists] = useState<
    Playlist[] | undefined
  >(undefined);
  const [modalError, setModalError] = useState<string | undefined>(undefined);

  const { isOpen: modalOpen, onOpen: onModalOpen, onClose } = useModal();

  const onModalClose = useCallback(() => {
    setModalError(undefined);
    onClose();
  }, [setModalError, onClose]);

  const getPlaylistData = useCallback(async () => {
    try {
      setLoading(true);
      const databasePlaylistsResponse = await getDatabasePlaylists();
      const spotifyPlaylistsResponse = await getSpotifyPlaylists();

      setMonitoredPlaylists(
        spotifyPlaylistsResponse.data
          .filter((spotifyPlaylist) =>
            databasePlaylistsResponse.data.some(
              (databasePlaylist) => databasePlaylist.id === spotifyPlaylist.id
            )
          )
          .map((monitoredPlaylist) => ({
            id: monitoredPlaylist.id,
            href: monitoredPlaylist.href,
            name: monitoredPlaylist.name,
            image: monitoredPlaylist.images[0],
          }))
      );

      setUnmonitoredPlaylists(
        spotifyPlaylistsResponse.data
          .filter(
            (spotifyPlaylist) =>
              !databasePlaylistsResponse.data.some(
                (databasePlaylist) => databasePlaylist.id === spotifyPlaylist.id
              )
          )
          .map((unmonitoredPlaylist) => ({
            id: unmonitoredPlaylist.id,
            href: unmonitoredPlaylist.href,
            name: unmonitoredPlaylist.name,
            image: unmonitoredPlaylist.images[0],
          }))
      );
      setLoading(false);
    } catch (e: any) {
      if (
        e?.response?.status === 500 &&
        e?.response?.data?.message ===
          "Application has not been logged into your Spotify account."
      ) {
        setShowSpotifyAuthModal(true);
      }
      setLoading(false);
    }
  }, [
    setLoading,
    getDatabasePlaylists,
    getSpotifyPlaylists,
    setMonitoredPlaylists,
    setUnmonitoredPlaylists,
  ]);

  useEffect(() => {
    getPlaylistData();
  }, [getPlaylistData]);

  const onSubmit = useCallback(
    async (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      setModalError(undefined);
      const playlistId = (event.currentTarget[1] as HTMLInputElement).value;
      const skipThreshold = (event.currentTarget[2] as HTMLInputElement).value;
      const ignoreInitialSkips = (event.currentTarget[3] as HTMLInputElement)
        .checked;
      const autoDeleteTracksAfter = (event.currentTarget[4] as HTMLInputElement)
        .value;

      if (!playlistId) {
        setModalError("Please select a playlist");
        return;
      }
      if (!!autoDeleteTracksAfter && parseInt(autoDeleteTracksAfter) === 0) {
        setModalError("Auto-delete must be empty or greater than 0");
        return;
      }

      try {
        setModalSaving(true);
        
        const request = {
          id: playlistId,
          skipThreshold: !!skipThreshold ? parseInt(skipThreshold) : undefined,
          ignoreInitialSkips: ignoreInitialSkips,
          autoCleanupLimit: !!autoDeleteTracksAfter
            ? parseInt(autoDeleteTracksAfter)
            : undefined,
        };

        await addDatabasePlaylist(request);
        setModalSaving(false);
        onModalClose();
        await getPlaylistData();
      } catch (e: any) {
        setModalError(e.response?.data?.message || "Unknown error");
        setModalSaving(false);
      }
    },
    [addDatabasePlaylist, getPlaylistData, onModalClose]
  );

  const onPlaylistChange = useCallback(
    async (newValue: { label: string; value: string }) => {
      if (newValue?.value) {
        setModalError(undefined);
      }
    },
    []
  );

  return {
    monitoredPlaylists,
    unmonitoredPlaylists,
    loading,
    modalOpen,
    onModalOpen,
    onModalClose,
    onSubmit,
    onPlaylistChange,
    modalError,
    modalSaving,
    showSpotifyAuthModal,
  } as const;
};
