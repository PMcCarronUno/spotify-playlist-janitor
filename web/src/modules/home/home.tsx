import styled from "styled-components";
import { AiOutlinePlusCircle } from "react-icons/ai";

import { BLACK, GREEN, RED } from "shared/constants";
import { SubTitle, Text } from "shared/components/typography";
import { Modal } from "shared/components/modal";

import { useHomeLogic } from "./use-home-logic";
import { AddPlaylistModalView } from "./modal/add-playlist-modal";

export const Home = () => {
  const {
    monitoredPlaylists,
    unmonitoredPlaylists,
    loading,
    modalOpen,
    onModalOpen,
    onModalClose,
    onSubmit,
    onPlaylistChange,
    modalError,
    modalSaving,
    showSpotifyAuthModal,
  } = useHomeLogic();

  return (
    <PageContainer>
      {monitoredPlaylists && (
        <>
          {monitoredPlaylists.map((monitoredPlaylist, index) => (
            <div
              key={monitoredPlaylist.id}
              className="item"
              onClick={() =>
                alert(
                  `Naviagte to ${monitoredPlaylist.name} at route ${monitoredPlaylist.id}`
                )
              }
              id={`playlist-item-${index}`}
              data-testid={`playlist-item-${index}`}
            >
              {monitoredPlaylist.image && (
                <img
                  alt={monitoredPlaylist.name}
                  width={130}
                  height={130}
                  src={monitoredPlaylist.image.url}
                />
              )}

              <SubTitle>{monitoredPlaylist.name}</SubTitle>
            </div>
          ))}
          <div
            className="item new"
            onClick={onModalOpen}
            id={`add-playlist-item`}
            data-testid={`add-playlist-item`}
          >
            <AiOutlinePlusCircle />
          </div>
        </>
      )}
      <Modal
        {...{
          isOpen: modalOpen,
          onClose: onModalClose,
          label: "Select Playlist Modal",
        }}
      >
        <AddPlaylistModalView
          {...{
            onSubmit,
            unmonitoredPlaylists,
            onPlaylistChange,
            modalSaving,
            modalError,
            onModalClose,
          }}
        />
      </Modal>
      <Modal
        {...{
          isOpen: showSpotifyAuthModal,
          onClose: () => {},
          label: "Spotify Auth Modal",
        }}
      >
        <ModalContainer>
          <SubTitle style={{ marginBottom: 8 }}>
            Spotify Authentication Error
          </SubTitle>
          <Text>
            The application has not been authenticated with your Spotify
            account. Please follow{" "}
            <a
              href={process.env.REACT_APP_API_URL}
              target="_blank"
              rel="noreferrer"
            >
              this
            </a>{" "}
            link and sign into Spotify, and <a href=".">reload</a> this page.
          </Text>
        </ModalContainer>
      </Modal>
    </PageContainer>
  );
};

const PageContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  align-content: flex-start;

  color: white;
  width: 100%;
  max-width: 1825px;
  paddingtop: 18px;

  .item {
    flex: 1 1 30%; /*grow | shrink | basis */
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    margin: 8px;
    height: 160px;
    max-width: 33%;
    background-color: ${BLACK};

    border-radius: 15px;

    cursor: pointer;

    &:hover {
      transform: scale(1.01);
      box-shadow: 0 0 32px rgba(117, 117, 117, 0.2);
    }
  }

  .new {
    background-color: ${GREEN};

    svg {
      height: 6em;
      width: 6em;
    }
  }
`;

const ModalContainer = styled.div`
  width: 360px;

  .rows {
    margin-bottom: 8px;
  }

  .row {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-bottom: 8px;
  }

  .number-input {
    width: 55px !important;
  }

  .toggle-input {
    margin-right: 16px;
  }

  .auto-delete-input {
    margin-right: -10px;
  }

  .auto-delete-post-text {
    margin-right: 10px;
  }

  .error {
    color: ${RED};
  }
`;

Home.displayName = "Home";
