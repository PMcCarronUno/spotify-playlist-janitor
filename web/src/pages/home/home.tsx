import styled from "styled-components";
import { HEADER_HEIGHT, HEADER_PADDING } from "shared/constants";
import { Home as HomeInitialise } from "modules/home";

export const Home = () => {
  return (
    <Content>
      <HomeInitialise />
    </Content>
  );
};

const Content = styled.div`
  display: flex;
  justify-content: center;
  height: calc(
    100vh - ${HEADER_HEIGHT} - ${HEADER_PADDING} - ${HEADER_PADDING}
  );
`;

Home.displayName = "Home";
