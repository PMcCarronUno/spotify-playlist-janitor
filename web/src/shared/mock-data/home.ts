import { Playlist } from "modules/home/home-types";
import { databasePlaylists, spotifyPlaylists } from "shared/mock-data/api";

export const monitoredPlaylists: Playlist[] = spotifyPlaylists
  .filter((spotifyPlaylist) =>
    databasePlaylists.some(
      (databasePlaylist) => databasePlaylist.id === spotifyPlaylist.id
    )
  )
  .map((monitoredPlaylist) => ({
    id: monitoredPlaylist.id,
    href: monitoredPlaylist.href,
    name: monitoredPlaylist.name,
    image: monitoredPlaylist.images[0],
  }));

export const unmonitoredPlaylists: Playlist[] = spotifyPlaylists
  .filter(
    (spotifyPlaylist) =>
      !databasePlaylists.some(
        (databasePlaylist) => databasePlaylist.id === spotifyPlaylist.id
      )
  )
  .map((monitoredPlaylist) => ({
    id: monitoredPlaylist.id,
    href: monitoredPlaylist.href,
    name: monitoredPlaylist.name,
    image: monitoredPlaylist.images[0],
  }));
