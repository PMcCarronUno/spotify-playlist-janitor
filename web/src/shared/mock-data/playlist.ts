import { DetailedPlaylist } from "modules/playlist/playlist-types";
import { databasePlaylists, spotifyPlaylists } from "shared/mock-data/api";
import { Image } from "shared/types";

const image: Image = {
  height: spotifyPlaylists.find(x => x.id === databasePlaylists[0].id)?.images[0]?.height || 100,
  width: spotifyPlaylists.find(x => x.id === databasePlaylists[0].id)?.images[0]?.width || 100,
  url: spotifyPlaylists.find(x => x.id === databasePlaylists[0].id)?.images[0]?.url || "href"
}

export const playlist: DetailedPlaylist = {
  id: databasePlaylists[0].id,
  href: spotifyPlaylists.find(x => x.id === databasePlaylists[0].id)?.href || "",
  name: spotifyPlaylists.find(x => x.id === databasePlaylists[0].id)?.name || "",
  image: image,
  ignoreInitialSkips: databasePlaylists[0].ignoreInitialSkips,
  autoCleanupLimit: databasePlaylists[0].autoCleanupLimit,
  skipThreshold: databasePlaylists[0].skipThreshold,
};
