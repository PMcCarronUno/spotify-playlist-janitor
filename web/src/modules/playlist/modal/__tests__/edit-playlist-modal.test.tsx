import { fireEvent, getByTestId, render } from "@testing-library/react";

import { AddPlaylistModalView } from "../edit-playlist-modal";
import { unmonitoredPlaylists } from "shared/mock-data/home";

describe("<AddPlaylistModalView />", () => {
  let container: HTMLElement;
  let props: any = {
    onSubmit: jest.fn(),
    unmonitoredPlaylists,
    onPlaylistChange: jest.fn(),
    modalSaving: false,
    modalError: undefined,
    onModalClose: jest.fn(),
  };

  beforeEach(async () => {
    ({ container } = render(<AddPlaylistModalView {...props} />));
  });

  it("should render AddPlaylistModalView component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render AddPlaylistModalView component with an error", () => {
    props = {
      ...props,
      modalError: "MODAL_ERROR",
    };

    ({ container } = render(<AddPlaylistModalView {...props} />));

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onSubmit function when Confirm button clicked", () => {
    const confirm = getByTestId(container, "confirm-modal-button");
    fireEvent.click(confirm);

    expect(props.onSubmit).toHaveBeenCalled();
  });

  it("should call onModalClose function when Cancel button clicked", () => {
    const cancel = getByTestId(container, "cancel-modal-button");
    fireEvent.click(cancel);

    expect(props.onModalClose).toHaveBeenCalled();
  });
});
