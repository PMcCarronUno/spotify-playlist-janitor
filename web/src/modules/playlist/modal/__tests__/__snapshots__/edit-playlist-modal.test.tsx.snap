// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPlaylistModalView /> should render EditPlaylistModalView component 1`] = `
<form
  autocomplete="off"
>
  <div
    class="sc-irLvoH fNqTLJ"
  >
    <div
      class="sc-guDMob jLqGJD"
      style="margin-bottom: 8px;"
    >
      Edit playlist details
    </div>
    <div
      class="rows"
    >
      <div
        class="row"
      >
        <div
          class="sc-dmyDmy dcXriK"
        >
          Skip threshold (seconds):
        </div>
        <div
          class="sc-fsYfxw lbnPWm"
          varaint="boxed"
        >
          <div
            class="input-container"
          >
            <input
              class="number-input"
              data-testid="skip-threshold-input"
              id="skip-threshold-input"
              max="999"
              min="1"
              type="number"
              value=""
            />
          </div>
        </div>
        <svg
          data-tooltip-id="skip-threshold-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="row"
      >
        <div
          class="toggle-input"
        >
          <label
            class="sc-qZruQ kOxPGx"
          >
            <span>
              Ignore initial skips
            </span>
            <input
              class="sc-kFCrIq fLrqdd"
              data-testid="ignore-intial-skips-toggle"
              id="ignore-intial-skips-toggle"
              type="checkbox"
            />
            <div
              class="sc-jsEeA-d jxYxBZ"
            />
          </label>
        </div>
        <svg
          data-tooltip-id="initial-skips-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="row"
      >
        <div
          class="sc-dmyDmy dcXriK"
        >
          Auto-delete tracks after:
        </div>
        <div
          class="sc-fsYfxw lbnPWm"
          varaint="boxed"
        >
          <div
            class="input-container"
          >
            <input
              class="number-input"
              data-testid="auto-delete-input"
              id="auto-delete-input"
              max="999"
              min="0"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="sc-dmyDmy dcXriK auto-delete-post-text"
        >
          skips
        </div>
        <svg
          data-tooltip-id="aut-delete-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: row; justify-content: flex-end;"
    >
      <button
        class="sc-jTQDJr sc-iBdoyZ gmttQv bbczBv secondary"
        data-testid="cancel-modal-button"
        id="cancel-modal-button"
        style="margin: 0px 4px;"
        type="button"
      >
        Cancel
      </button>
      <button
        class="sc-jTQDJr sc-gLLuHO gmttQv lckOnr primary"
        data-testid="confirm-modal-button"
        id="confirm-modal-button"
        style="margin: 0px 4px;"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</form>
`;

exports[`<EditPlaylistModalView /> should render EditPlaylistModalView component with an error 1`] = `
<form
  autocomplete="off"
>
  <div
    class="sc-irLvoH fNqTLJ"
  >
    <div
      class="sc-guDMob jLqGJD"
      style="margin-bottom: 8px;"
    >
      Edit playlist details
    </div>
    <div
      class="rows"
    >
      <div
        class="row"
      >
        <div
          class="sc-dmyDmy dcXriK"
        >
          Skip threshold (seconds):
        </div>
        <div
          class="sc-fsYfxw lbnPWm"
          varaint="boxed"
        >
          <div
            class="input-container"
          >
            <input
              class="number-input"
              data-testid="skip-threshold-input"
              id="skip-threshold-input"
              max="999"
              min="1"
              type="number"
              value=""
            />
          </div>
        </div>
        <svg
          data-tooltip-id="skip-threshold-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="row"
      >
        <div
          class="toggle-input"
        >
          <label
            class="sc-qZruQ kOxPGx"
          >
            <span>
              Ignore initial skips
            </span>
            <input
              class="sc-kFCrIq fLrqdd"
              data-testid="ignore-intial-skips-toggle"
              id="ignore-intial-skips-toggle"
              type="checkbox"
            />
            <div
              class="sc-jsEeA-d jxYxBZ"
            />
          </label>
        </div>
        <svg
          data-tooltip-id="initial-skips-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="row"
      >
        <div
          class="sc-dmyDmy dcXriK"
        >
          Auto-delete tracks after:
        </div>
        <div
          class="sc-fsYfxw lbnPWm"
          varaint="boxed"
        >
          <div
            class="input-container"
          >
            <input
              class="number-input"
              data-testid="auto-delete-input"
              id="auto-delete-input"
              max="999"
              min="0"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="sc-dmyDmy dcXriK auto-delete-post-text"
        >
          skips
        </div>
        <svg
          data-tooltip-id="aut-delete-tooltip"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 1a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zm0 12a5.5 5.5 0 1 1 0-11 5.5 5.5 0 0 1 0 11zm1.55-8.42a1.84 1.84 0 0 0-.61-.42A2.25 2.25 0 0 0 7.53 4a2.16 2.16 0 0 0-.88.17c-.239.1-.45.254-.62.45a1.89 1.89 0 0 0-.38.62 3 3 0 0 0-.15.72h1.23a.84.84 0 0 1 .506-.741.72.72 0 0 1 .304-.049.86.86 0 0 1 .27 0 .64.64 0 0 1 .22.14.6.6 0 0 1 .16.22.73.73 0 0 1 .06.3c0 .173-.037.343-.11.5a2.4 2.4 0 0 1-.27.46l-.35.42c-.12.13-.24.27-.35.41a2.33 2.33 0 0 0-.27.45 1.18 1.18 0 0 0-.1.5v.66H8v-.49a.94.94 0 0 1 .11-.42 3.09 3.09 0 0 1 .28-.41l.36-.44a4.29 4.29 0 0 0 .36-.48 2.59 2.59 0 0 0 .28-.55 1.91 1.91 0 0 0 .11-.64 2.18 2.18 0 0 0-.1-.67 1.52 1.52 0 0 0-.35-.55zM6.8 9.83h1.17V11H6.8V9.83z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="row error"
        data-testid="modal-error"
        id="modal-error"
      >
        <div
          class="sc-hLQTFJ gjIzVE"
        >
          MODAL_ERROR
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: row; justify-content: flex-end;"
    >
      <button
        class="sc-jTQDJr sc-iBdoyZ gmttQv bbczBv secondary"
        data-testid="cancel-modal-button"
        id="cancel-modal-button"
        style="margin: 0px 4px;"
        type="button"
      >
        Cancel
      </button>
      <button
        class="sc-jTQDJr sc-gLLuHO gmttQv lckOnr primary"
        data-testid="confirm-modal-button"
        id="confirm-modal-button"
        style="margin: 0px 4px;"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</form>
`;
